docs/docs/reference/02-yaml-syntax.md

```diff
@@ -0,0 +1,638 @@
+---
+sidebar_position: 1
+---
+
+# YAML application syntax
+
+The following sections define the YAML syntax used to describe an AICA application.
+
+## Overview
+
+An application description contains the following top-level elements.
+
+```yaml
+on_start:
+  ...
+
+hardware_interfaces:
+  ...
+
+conditions:
+  ...
+
+components:
+  ...
+```
+
+## On Start
```

Comment:
Do we need to update the docs for yaml syntax? for example, we have buttons now in the application.yml

Reply:
Good point, buttons are not documented. Because buttons are a graph / UI feature, and this is a programming reference for developers who might just want to use the YAML syntax on its own, I will leave that as a TODO.